ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/AbstractTransactionsLayer.java

```diff
@@ -91,6 +93,11 @@ protected AbstractTransactionsLayer(
     metrics.initSpaceUsed(this::getLayerSpaceUsed, name());
     metrics.initTransactionCount(pendingTransactions::size, name());
     metrics.initUniqueSenderCount(txsBySender::size, name());
+    Arrays.stream(TransactionType.values())
+        .forEach(
+            type ->
+                metrics.initTransactionCountByType(
+                    () -> txCountByType[type.ordinal()], name(), type));
```

Comment:
using ordinal to track the external feels fragile, but if experimental tx types start showing up they may leave large gaps.  However... it's what we express externally, so this looks like it will work.  See my other comment on the metric.